### PR TITLE
WIP: add dbus interface

### PR DESCRIPTION
**Pull request by chmouel**
Add a simple dbus interface to call shortcuts from external, works like
this :

```
gdbus call --session --dest=org.gnome.Shell --object-path /org/gnome/Shell/Extensions/RunOrRaise --method 'org.gnome.Shell.Extensions.RunOrRaise.SendShortcut' "<Super>f"
```

will trigger the super+f keys as configured in config file

this is still a WIP, request for comment, as probably need some refactoring for config parsing

fix issues #40 

Signed-off-by: Chmouel Boudjnah <chmouel@chmouel.com>

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -2,10 +2,21 @@ const Main = imports.ui.main;
 const Meta = imports.gi.Meta;
 const Shell = imports.gi.Shell;
 const Me = imports.misc.extensionUtils.getCurrentExtension();
-const {spawnCommandLine, spawn} = imports.misc.util;
+const { spawnCommandLine, spawn } = imports.misc.util;
```
Comment (chmouel) [inline extension.js:5]: I think lsp auto formatter had kicked in without me knowing (i don't do much js), i'll remove them if needed.

Comment (e3rd) [reply extension.js:5]: Yes please, remove them. I'm using ESLint in PyCharm that would tend to restore them back.

**Comment by e3rd** [comment]
Hi,
Thanks for the contribution. I am not contrary to such features so long as it does not affect performance. As I have no experience with dbus you can freely continue with the development. (No code duplication in the final phase please.)

Please note the project should fulfill some tasks #42 given by a Gnome administrator before a new version might be released. (If you come along some of them while tackling the dbus, I'll be glad.)

**Comment by chmouel** [comment]
@e3rd hello edvard, thanks for your detailed answer, I don't think dbus interface have any impact on performances, except if a bad actor start to abuse it.

but more i think about it and more i think it may be maybe more fitting to have this in a dedicated extensions to be able to extend to other action for keyboard integration on gnome that are not just run or focus (like windows management too).

I don't think we need to bloat your extension for this,

thanks a lot again and this list you have on your issue looks very useful for anyone contributing on ext.gnome.org

**Comment by e3rd** [comment]
Thanks for considering though! :)

**Pull request closed without merging**